### PR TITLE
Run all tests on CoreCLR

### LINKED ISSUE #3781: Run all tests on .NET Core
Initial work on this work: https://github.com/natemcmaster/EntityFramework/tree/namc/tfm
Once we have NetCore version of Moq, we can build and run tests on dnxcore.
See https://github.com/castleproject/Core/issues/90

### PR DESCRIPTION
**Pull request by natemcmaster**
Run **all** tests on CoreCLR, not just functional tests. 

Fix #3781

### CODE REVIEW

File: test/EntityFramework.MicrosoftSqlServer.Tests/SqlServerDatabaseCreatorTest.cs [hunk 1]
```diff
@@ -220,19 +221,38 @@ public Task<object> ExecuteScalarAsync(IRelationalConnection connection, Cancell
...
+            }
+
+            var errors = (SqlErrorCollection)typeof(SqlErrorCollection)
+                .GetTypeInfo()
+                .DeclaredConstructors
+                .Single()
+                .Invoke(null);
 
-            typeof(SqlErrorCollection).GetTypeInfo().GetRuntimeMethods().Single(m => m.Name == "Add").Invoke(errors, new object[] { error });
+            typeof(SqlErrorCollection).GetRuntimeMethods().Single(m => m.Name == "Add").Invoke(errors, new object[] { error });
```
Comment (AndriySvyryd) [inline test/EntityFramework.MicrosoftSqlServer.Tests/SqlServerDatabaseCreatorTest.cs]: What's the reason to not use `GetTypeInfo()` ?

**Comment by AndriySvyryd** [comment]
:shipit:

File: test/EntityFramework.MicrosoftSqlServer.Tests/SqlServerDatabaseCreatorTest.cs (hunk 1 above)
Comment (natemcmaster) [inline test/EntityFramework.MicrosoftSqlServer.Tests/SqlServerDatabaseCreatorTest.cs]: `TypeInfo.GetRuntimeMethods()` does not exist in .NET Core. The same result could have been be done with `GetTypeInfo().DeclaredMethods`

File: test/EntityFramework.Core.Tests/ApiConsistencyTestBase.cs [hunk 2]
```diff
@@ -53,8 +53,8 @@ public void Public_api_arguments_should_have_not_null_annotation()
         {
             var parametersMissingAttribute
                 = (from type in GetAllTypes(TargetAssembly.GetTypes())
-                   where type.IsVisible && !typeof(Delegate).GetTypeInfo().IsAssignableFrom(type)
-                   let interfaceMappings = type.GetInterfaces().Select(type.GetInterfaceMap)
+                   where type.GetTypeInfo().IsVisible && !typeof(Delegate).GetTypeInfo().IsAssignableFrom(type.GetTypeInfo())
+                   let interfaceMappings = type.GetInterfaces().Select(i => type.GetTypeInfo().GetRuntimeInterfaceMap(i))
```
Comment (smitpatel) [inline test/EntityFramework.Core.Tests/ApiConsistencyTestBase.cs]: Thanks for this! :smile:

**Comment by natemcmaster** [comment]
Blocked on issue in OSX 10.9 https://github.com/dotnet/corefx/pull/4729.

**Pull request merged**